transformer/classes/dockerfile/pythondockerfilegenerator.go

```diff
@@ -0,0 +1,213 @@
+	"github.com/konveyor/move2kube/environment"
+	"github.com/konveyor/move2kube/types/qaengine/commonqa"
+	transformertypes "github.com/konveyor/move2kube/types/transformer"
+	"github.com/konveyor/move2kube/types/transformer/artifacts"
+	"github.com/sirupsen/logrus"
+)
+
+const (
+	pythonExt           = ".py"
+	pythonMain          = "main"
+	pythonManage        = "manage"
+	pythonService       = "python"
+	djangoService       = "django"
+	requirementsTxtFile = "requirements.txt"
+	// MainPythonFilePathType points to the .py file path which contains main function
+	MainPythonFilePathType transformertypes.PathType = "MainPythonFileRelPath"
+)
+
+var (
+	pythonMainRegex = regexp.MustCompile(`__main__`)
+)
+
+// PythonDockerfileGenerator implements the Transformer interface
+type PythonDockerfileGenerator struct {
+	Config transformertypes.Transformer
+	Env    *environment.Environment
+}
+
+// PythonTemplateConfig implements python config interface
+type PythonTemplateConfig struct {
+	AppName           string
+	Port              int32
+	MainScriptRelPath string
+	RequirementsTxt   string
+	IsDjangoProject   bool
+}
+
+// Init Initializes the transformer
+func (t *PythonDockerfileGenerator) Init(tc transformertypes.Transformer, env *environment.Environment) (err error) {
+	t.Config = tc
+	t.Env = env
+	return nil
+}
+
+// GetConfig returns the transformer config
+func (t *PythonDockerfileGenerator) GetConfig() (transformertypes.Transformer, *environment.Environment) {
+	return t.Config, t.Env
+}
+
+// findMainScript returns the path of .py file having the main function
+func findMainScript(pythonFilesPath []string) (string, error) {
+	if len(pythonFilesPath) == 0 {
+		return "", nil
+	}
+	for _, pythonFilePath := range pythonFilesPath {
+		pythonFile, err := os.Open(pythonFilePath)
+		if err != nil {
+			logrus.Debugf("Failed to open the file %s", pythonFilePath)
+		}
+		scanner := bufio.NewScanner(pythonFile)
+		scanner.Split(bufio.ScanLines)
+		for scanner.Scan() {
+			if pythonMainRegex.MatchString(scanner.Text()) {
+				return pythonFilePath, nil
+			}
+		}
+	}
+	return "", fmt.Errorf("could not find the main function in python files")
+}
+
+// DirectoryDetect runs detect in each sub directory
+func (t *PythonDockerfileGenerator) DirectoryDetect(dir string) (services map[string][]transformertypes.Artifact, err error) {
+	pythonFiles, err := common.GetFilesByExt(dir, []string{pythonExt})
+	if err != nil {
+		logrus.Errorf("Error while finding python files %s", err)
+		return nil, nil
+	}
+	if len(pythonFiles) == 0 {
+		return nil, nil
+	}
+	mainPythonFilePath, err := findMainScript(pythonFiles)
+	if err != nil {
+		return nil, err
+	}
+	// Search requirements.txt
+	if mainPythonFilePath != "" {
+		serviceName := strings.TrimSuffix(filepath.Base(mainPythonFilePath), pythonExt)
+		if serviceName == pythonMain {
+			serviceName = pythonService
+		} else if serviceName == pythonManage {
+			serviceName = djangoService
```

Comment:
In general, what I noticed was that all the django projects have this `manage.py` file, so I was using that for deciding whether it's a django project or not. But instead of doing that, now we are using the `requirements.txt` file and checking if `django` is present in it or not.